### PR TITLE
Add info about other transaction methods

### PR DESCRIPTION
**Pull request by MongoCaleb**
### Jira

- https://jira.mongodb.org/browse/DOCSP-17717

### Staged Changes (Requires MongoDB Corp SSO)

- https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-17717/sdk/dotnet/fundamentals/write-transactions/

### CODE REVIEW

**Review by cbush — COMMENTED** [review]
Commenting until seeing reply on the Jira comment thread. I'm ok with pointing out the existence of the methods, but I don't think its usage should be encouraged. This change seems alright for what it is.

That said, I find trying to explain this concept adds more complexity that we may prefer to avoid. Frankly, using transactions manually (i.e. BeginWrite()) looks like a bad practice to me, and we've taken that stance with the other SDKs.

File: examples/dotnet/Examples/Writes.cs [hunk 1]
```diff
@@ -42,19 +55,43 @@ public void Foo()
...
+            {
+                // Do other work that needs to be included in
+                // this transaction
+            }
+            catch (Exception ex)
+            {
+                // Something went wrong; roll back the transaction
+                transaction.Dispose();
+            }
+            transaction.Commit();
```
[the remark this answers is not in the document]
Comment (MongoCaleb) [reply examples/dotnet/Examples/Writes.cs]: Fixed

**Comment by papafe** [comment]
I think in general the doc looks fine. What I would add is the fact that the transaction returned by `BeginWrite()` implements IDisposable, so it can be used like this (but not necessarily):

```
using(var transaction = realm.BeginWrite())
{
    person.FirstName = "John";
    person.Age = 56;
    transaction.Commit();
}
```

Also I am not sure why in the example for `BeginWrite` there is a try-catch block but `realm.Add(myDog)` is outside. That should probably be inside the block, or (even better in my opinion) the try catch should be completely removed, as it seems like `BeginWrite` is more "dangerous" than `Write`. In reality `Write` is just using `BeginWrite` inside.

**Comment by papafe** [comment]
Mmh, I see your point! I think in this case it makes sense to have the try-catch, but `realm.Add(myDog)` should be inside the block for sure.

**Review by cbush — APPROVED** [review]

**Pull request merged**